dropwizard-jdbi/src/main/java/io/dropwizard/jdbi/DBIFactory.java

```diff
@@ -53,7 +57,14 @@ public DBI build(Environment environment,
         final String validationQuery = configuration.getValidationQuery();
         final DBI dbi = new DBI(dataSource);
         environment.lifecycle().manage(dataSource);
-        environment.healthChecks().register(name, new DBIHealthCheck(dbi, validationQuery));
+        ExecutorService executorService = environment.lifecycle().executorService("DBIHealthCheck")
```

Comment:
This is a good point. If we use CallerRunsPolicy, we will loose the ability to terminate the thread, so we end up with the original situation. Instead of being a single a threaded executor, what about creating a pool shared across health checks where the number of threads can be configured. We can still make it have a CallerRunsPolicy, but it will be less likely to hit it.